Source/Core/Duality/Utility/Coroutines/CoroutineManager.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Duality.Utility.Coroutines
+{
+	/// <summary>
+	/// The Coroutine Manager
+	/// </summary>
+	public class CoroutineManager
+	{
+		private readonly Queue<Coroutine> pool = new Queue<Coroutine>(64);
+		
+		private Queue<Coroutine> currentCycle = new Queue<Coroutine>(256);
+		private Queue<Coroutine> nextCycle = new Queue<Coroutine>(256);
+
+		private readonly Dictionary<Coroutine, Exception> lastFrameErrors = new Dictionary<Coroutine, Exception>();
+
+		/// <summary>
+		/// Returns an IEnumerable of all currently active and scheduled Coroutines
+		/// </summary>
+		public IEnumerable<Coroutine> Coroutines
+		{
+			get { return this.currentCycle.Concat(this.nextCycle); }
+		}
+
+		/// <summary>
+		/// Returns an IEnumerable of the Exception encountered in the last cycle, with the relative Coroutine
+		/// </summary>
+		public IReadOnlyDictionary<Coroutine, Exception> LastFrameErrors
+		{
+			get { return this.lastFrameErrors; }
+		}
+
+		/// <summary>
+		/// Prepares a new Coroutine and places it in the scheduled queue, to be started in the next cycle
+		/// </summary>
+		/// <param name="enumerator">The Coroutine's execution body</param>
+		/// <param name="name">The name of the Coroutine</param>
+		/// <returns>The prepared Coroutine</returns>
+		internal Coroutine StartNew(IEnumerable<WaitUntil> enumerator, string name)
```

Comment:
Would make this public, mostly useful for testing but maybe there will be also scenarios outside of a scene in the future